### PR TITLE
Bump <Textarea /> and <TextareaField /> to memo + forwardRef

### PR DESCRIPTION
**Pull request by akleiner2**
## Overview 
The title of this PR -- keeping chugging along converting stuff to forwardRef + memo + hooks.
 
## Screenshots (if applicable) 
N/A no visual change, just a component change.

## Testing
- [x] Updated Typescript types and/or component PropTypes 
- [x] Added / modified component docs 
- [x] Added / modified Storybook stories

### CODE REVIEW

File: src/textarea/src/Textarea.js [hunk 1]
```diff
@@ -1,100 +1,32 @@
...
       required,
-      isInvalid,
-      appearance,
       placeholder,
-      spellCheck,
-      grammarly,
-      ...props
-    } = this.props
+      height,
+      width = 280,
```
Comment (mshwery) [inline src/textarea/src/Textarea.js:20]: Why the width default value change (from `100%`)?

File: src/textarea/src/Textarea.js [hunk 2]
```diff
@@ -114,11 +46,71 @@ class Textarea extends PureComponent {
...
+
+  /**
+   * The width of the TextInput.
+   */
+  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+  /**
+   * Theme provided by ThemeProvider.
+   */
+  theme: PropTypes.object.isRequired,
```
Comment (mshwery) [inline src/textarea/src/Textarea.js:107]: Can remove!

File: src/textarea/src/TextareaField.js [hunk 3]
```diff
@@ -116,9 +61,56 @@ export default class TextareaField extends PureComponent {
           appearance={appearance}
           placeholder={placeholder}
           spellCheck={spellCheck}
+          ref={ref}
```
Comment (mshwery) [inline src/textarea/src/TextareaField.js:64]: Technically for backwards compat, shouldn't the ref be on the outer component?

**Comment by akleiner2** [comment]
Think this was actually superseded by the codemod in #825 so I'll just keep it as is! Too fast.

**Pull request closed without merging**